### ISSUE TITLE
Support libsql

### ISSUE THREAD

**Issue by rawkode** [go-gitea/gitea#29542]
### Feature Description

I for love to see the SQLite driver swapped out for libsql, allowing us to use a highly available sqlite database. 

This should be close to a drop in replacement.

**Comment by delvh** [comment]
Sounds good to me.
Only the
> To get started with the libSQL API:
    [Go](https://github.com/tursodatabase/libsql/blob/main/bindings/go) (experimental)

is a bit concerning to me

**Comment by lunny** [comment]
Sorry, maybe we need some investigate whether libsql is mature enough.

**Comment by delvh** [comment]
I mean, according to their docs, they are a fork of SQLite.
In other words, they are as mature as SQLite.

**Comment by silverwind** [comment]
Does this Go binding require CGO?

**Comment by delvh** [comment]
Probably

**Comment by techknowlogick** [comment]
Yes, it does require CGO. I've just thrown up an experimental branch

**Comment by rawkode** [comment]
There's a client which I don't believe needs CGO

https://github.com/tursodatabase/libsql-client-go

**Comment by rawkode** [comment]
However, I believe that will only support libsql and not SQLite
